plugins/inputs/mesos/mesos.go

```diff
@@ -95,6 +95,14 @@ func (m *Mesos) SetDefaults() {
 		log.Println("I! [mesos] Missing timeout value, setting default value (100ms)")
 		m.Timeout = 100
 	}
+	if client.Transport == nil {
```

Comment:
If `client.Transport` is a singleton then why not set it down at line (409) where we are initializing `client`.